### ISSUE TITLE
False positive with certain if-else constructs

### ISSUE THREAD

**Issue by cespare** [golang/lint#203]
If you run golint on the following file:

``` go
package main

import "errors"

func main() {
}

func f() (int, error) {
	var a, b, c bool
	var n int
	if a {
		n = 1
	} else if b {
		n = 2
	} else if c {
		return 0, errors.New("foo")
	} else { // line 17
		return 0, errors.New("bar")
	}
	return n, nil
}
```

it gives

    test.go:17:9: if block ends with a return statement, so drop this else and outdent its block

but this isn't correct, because some of the if-else arms don't end in `return`.

It doesn't give this false positive if you remove the first else-if:

``` go
func f() (int, error) {
	var a, b, c bool
	var n int
	if a {
		n = 1
	} else if c {
		return 0, errors.New("foo")
	} else {
		return 0, errors.New("bar")
	}
	return n, nil
}
```

**Comment by dsymonds** [comment]
That looks like a bug. I thought it had already been fixed, but perhaps not.

**Comment by dsymonds** [comment]
Oh, this is #181.

**Comment by cespare** [comment]
Ah, thanks. I'd only checked the open issues, not PRs.

**Comment by JKobyP** [comment]
Since this hasn't been fixed, should it be reopened?

**Comment by cespare** [comment]
This is a duplicate of #181. No need to keep multiple issues open for the same bug.

**Comment by JKobyP** [comment]
#181 is a PR, not an issue
